optuna_integration/comet/comet.py

```diff
@@ -0,0 +1,207 @@
+from typing import Union
+
+import optuna
+from optuna.study.study import ObjectiveFuncType
+
+from optuna_integration._imports import try_import
+
+
+with try_import() as _imports:
+    import comet_ml
+
+
+class CometCallback:
+    """
+    A callback for logging Optuna study trials to a Comet ML Experiment. Comet ML must be installed to run.
+
+    This callback is intended for use with Optuna's study.optimize() method. It ensures that all trials
+    from an Optuna study are logged to a single Comet Experiment, facilitating organized tracking
+    of hyperparameter optimization. The callback supports both single and multi-objective optimization.
+
+    In a distributed training context, where trials from the same study might occur on different machines,
+    this callback ensures consistency by logging to the same Comet Experiment using an experiment key
+    stored within the study's user attributes.
+
+    By default, Trials are logged as Comet Experiments, which will automatically log code, system metrics,
+    and many other values. However, it also adds some computational overhead (potentially a few seconds).
+
+    Parameters:
+    - study (optuna.study.Study): The Optuna study object to which the callback is attached.
+    - workspace (str) - Optional: The workspace in Comet ML where the project resides.
+    - project_name (str) - Optional: The name of the project in Comet ML where the experiment will be logged. Defaults to "general"
+    - metric_names ([str]) - Optional: A list of the names of your objective metrics.
+
+    Usage:
+    ```
+    study = optuna.create_study(directions=["minimize", "maximize"])
+    comet_callback = CometCallback(study, metric_names=["accuracy", "top_k_accuracy"],
+                                     project_name="your_project_name", workspace="your_workspace")
+    study.optimize(your_objective_function, n_trials=100, callbacks=[comet_callback])
+    ```
+
+    Note:
+    The callback checks for an existing Comet Experiment key in the study's user attributes. If present, it initializes
+    an ExistingExperiment; otherwise, it creates a new APIExperiment and stores its key in the study for future reference.
+
+    You will need a Comet API key to log data to Comet.
+
+    You can also log extra data directly to your Trial's Experiment via the objective function by using
+    the @CometCallback.track_in_comet decorator, which exposes an `experiment` property on your trial, like so:
+    ```
+    study = optuna.create_study(directions=["minimize", "maximize"])
+    comet_callback = CometCallback(study, metric_names=["accuracy", "top_k_accuracy"],
+                                     project_name="your_project_name", workspace="your_workspace")
+
+    @comet_callback.track_in_comet()
+    def your_objective(trial):
+        trial.experiment.log_other("foo", "bar")
+        # Rest of your objective function...
+
+    study.optimize(your_objective, n_trials=100, callbacks=[comet_callback])
```

Comment:
Could you follow the same format of other integration such as https://github.com/optuna/optuna-integration/blob/main/optuna_integration/wandb/wandb.py for documentation?